### DISCUSSION TITLE
form.reset call inside useEffect

### DISCUSSION THREAD

**Discussion by popuguytheparrot** [react-hook-form/react-hook-form#7720] [Q&A] [answered]
Hi. How to right call form.reset on component mount without loop useEffect with deps [form] ? 

```jsx
useEffect(() => {
        const innerParams = params;
        formContext.reset(innerParams);
    }, [formContext, params]);
```

**Comment by Moshyfawn** [comment, marked as answer]
Hey!
Usually it's as simple as doing this:
```js
useEffect(() => {
  if (data) {
    reset(data);
  }
}, [reset, data]);
```
But, I guess, depends on your use-case.
See [this CSB](https://codesandbox.io/s/agitated-forest-couu7?file=/src/App.tsx) for example
